opensafely/codelists.py

```diff
@@ -18,6 +19,10 @@
 MANIFEST_FILE = "codelists.json"
 
 
+def request_headers():
+    return {"User-Agent": f"OpenSAFELY-CLI/{opensafely.__version__.lstrip('v')}"}
```

Comment:
```suggestion
HEADERS =  {"User-Agent": f"OpenSAFELY-CLI/{opensafely.__version__.lstrip('v')}"}
```

I'm not sure if this needed to become a function rather than a module-level variable as it was before, which is a little simpler? I don't think the `opensafely` module can change version during the lifetime of this module, as it's only imported once.

Reply:
if it's a module-level variable a partially-initialised module `AttributeError` is raised. Wrapping it in a function that is called when everything is seemingly properly initialised (as per [`upgrade`](https://github.com/opensafely-core/opensafely-cli/blob/61c0e7411bdbf00d739b53e843e164dc3bb90dfc/opensafely/upgrade.py#L86C8-L89C9)) works

Reply:
Oh I see, this PR has introduced a circular import. `opensafely/__init__.py` imports `opensafely/codelists.py` which now imports `opensafely`. I suppose we should avoid that if we can do so easily enough.

Can the versioning stuff move out of `opensafely/__init__.py` into a separate module which then gets imported into `opensafely/__init__.py`, and `opensafely/codelists.py` import directly from that module, rather than circularly from `opensafely`?

Reply:
I'm inclined to put that refactoring into its own ticket